frame/assets/src/lib.rs

```diff
@@ -151,6 +151,16 @@ pub trait Config: frame_system::Config {
 	/// supports.
 	type AssetDepositPerZombie: Get<BalanceOf<Self>>;
 
+	/// The maximum length of a name or symbol stored on-chain.
+	type StringLimit: Get<usize>;
```

Comment:
it's fine in this instance (since it's not being stored anywhere and i very much doubt this would ever be set to something like `u32::max_value() + 1`), but i generally avoid potentially variable-size datatypes like `usize` from within consensus code, and instead just use `u32` (which all sensible platforms can store in a `usize`) and cast.

Reply:
ive updated. you are right, u32 is more sensible.